### PR TITLE
precompiles: Optionally use silkpre as implementation

### PR DESCRIPTION
**Pull request by chfast**
Add CMake option (off by default) `EVMONE_PRECOMPILES_SILKPRE` to enable building of silkpre as the precompiles implementation.

### CODE REVIEW

**Comment by chfast** [comment]
A Clang build of silkpre does not work. I'm getting `SIGEMT`.

**Comment by yperbasis** [comment]
> A Clang build of silkpre does not work. I'm getting `SIGEMT`.

Not sure if it's relevant, but perhaps https://github.com/torquem-ch/silkpre/pull/8 can help.

**Comment by chfast** [comment]
> > A Clang build of silkpre does not work. I'm getting `SIGEMT`.
> 
> Not sure if it's relevant, but perhaps [torquem-ch/silkpre#8](https://github.com/torquem-ch/silkpre/pull/8) can help.

This seems coming from libff.

File: test/state/precompiles_silkpre.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace
+{
+ExecutionResult execute(const uint8_t* input, size_t input_size, uint8_t* output_buf,
+    [[maybe_unused]] size_t max_output_size, PrecompileId id) noexcept
+{
+    const auto index = static_cast<size_t>(id) - 1;
+    const auto [output, output_size] = kSilkpreContracts[index].run(input, input_size);
+    if (output == nullptr)
+        return {EVMC_PRECOMPILE_FAILURE, 0};
+    assert(output_size <= max_output_size);
```
Comment (gumb0) [inline test/state/precompiles_silkpre.cpp:21]: Why is this an assertion? should you copy `max(max_output_size, output_size)` bytes?

File: test/state/precompiles.cpp [hunk 2]
```diff
@@ -172,6 +176,16 @@ inline constexpr auto traits = []() noexcept {
         {ecpairing_analyze, dummy_execute<PrecompileId::ecpairing>},
         {blake2bf_analyze, dummy_execute<PrecompileId::blake2bf>},
     }};
+#ifdef EVMONE_PRECOMPILES_SILKPRE
```
[the remark this answers is not in the document]
Comment (chfast) [reply test/state/precompiles.cpp:179]: This is structured as optional override. It also allows us quickly change what we override with silkpre.

File: test/state/precompiles_silkpre.cpp (hunk 1 above)
Comment (chfast) [reply test/state/precompiles_silkpre.cpp:21]: The `max_output_size` is computed by evmone's analysis and it should match what silkpre computes. But using `max` sounds good while keeping the assert.

File: test/state/precompiles_stub.json [hunk 3]
```diff
@@ -668,6 +668,7 @@
   },
   {
     "0537dab8bcfa337395fdfbd203d054ac25c08722e9416a0ac259b779a2c87721": "ba80a53f981c4d0d6a2797b69f12f6e94c212f14685ac4b74b12bb6fdbffa2d17d87c5392aab792dc252d5de4533cc9518d38aa8dbf1925ab92386edd4009923",
+    "05bd989b8f157e23dbb02b8bc54d344a60558de44a7ee3f804cf1dfc9e82bbda": "fc59093aafa9ab43daae0e914c57635c5402d8e3d2130eb9b3cc181de7f0ecf9b22bf99a7815ce16419e200e01846e6b5df8cc7703041bbceb571de6631d2615",
```
[the remark this answers is not in the document]
Comment (chfast) [reply test/state/precompiles_stub.json:671]: This is from one tests which we don't run by default (because it takes too long time). See commit message.

**Review by rodiazet — APPROVED** [review]

File: test/state/precompiles_silkpre.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+{
+    const auto index = stdx::to_underlying(id) - 1;
+    const auto [output, output_size] = kSilkpreContracts[index].run(input, input_size);
+    if (output == nullptr)
+        return {EVMC_PRECOMPILE_FAILURE, 0};
+
+    // Check if max_output_size computed by analysis match the computed result.
+    assert(output_size <= max_output_size);
+
+    auto trimmed_output_size = std::min(output_size, max_output_size);
```
Comment (gumb0) [inline test/state/precompiles_silkpre.cpp:25]:
```suggestion
    const auto trimmed_output_size = std::min(output_size, max_output_size);
```

**Review by gumb0 — APPROVED** [review]

**Pull request merged**